### PR TITLE
fix #768 issue (Manager.off method)

### LINKED ISSUE #768: Badly initialized object
The following code produces an error:

    (new Hammer(document.body)).off('swipeleft', function(){})
    //=> TypeError: Cannot read property 'splice' of undefined

### PR DESCRIPTION
**Pull request by sculove**
I resend pr that seperate unrelated changes.

When you didn't attach an event and you use a Hammerjs.Manager.off method , you will see script error.
I think that hammerjs have to handle exception.

I modified Hammerjs.Manager.off method.
also I added test case in test_events.js

- Related issue #768 
- Related pr #770

### CODE REVIEW

File: src/manager.js [hunk 1]
```diff
@@ -203,10 +203,12 @@ Manager.prototype = {
     off: function(events, handler) {
         var handlers = this.handlers;
         each(splitStr(events), function(event) {
-            if (!handler) {
-                delete handlers[event];
-            } else {
-                handlers[event].splice(inArray(handlers[event], handler), 1);
+            if (typeof handlers[event] !== 'undefined') {
```
Comment (arschmitz) [inline src/manager.js]: do we really need to check `typeof handlers[event] !== 'undefined'` here isn't it enough to just check `handlers[event]` ?

File: tests/unit/test_events.js [hunk 2]
```diff
@@ -47,3 +47,15 @@ test('test the eventemitter', function() {
...
+ * Hammer.Manager.off method : exception handling
+ */
+test("When Hammer.Manager didn't attach an event, 'off' method is ignored", function() {
+    var count = 0;
+    hammer = new Hammer(el, { inputTarget: document.body });
+    hammer.off("swipeleft", function(e) {
+        count++;
+    });
+    ok(true, "nothing");
+});
```
Comment (arschmitz) [inline tests/unit/test_events.js]: Missing new line at end of file

**Comment by arschmitz** [comment]
@sculove 2 small changes i would like to see but looks good thank you.

File: src/manager.js (hunk 1 above)
Comment (sculove) [inline src/manager.js]: your opinion is better. ok. I will change code.
how about following code?
```
if (!handler) {
    delete handlers[event];
} else {
    handlers[event] && handlers[event].splice(inArray(handlers[event], handler), 1);
}
```

File: tests/unit/test_events.js (hunk 2 above)
Comment (sculove) [inline tests/unit/test_events.js]: OK. I got it

**Comment by sculove** [comment]
@arschmitz Thank you :)

**Pull request closed without merging**